lib/posix/clock.c

```diff
@@ -34,6 +45,9 @@ int clock_gettime(clockid_t clock_id, struct timespec *ts)
 		break;
 
 	case CLOCK_REALTIME:
+		if (rtc_dev) {
```

Comment:
I suspect this is OK, but somebody might want to make it optional, especially if the RTC maintains whole-second resolution.

Reply:
Indeed, I can use a different option for that like CONFIG_USE_RTC_FOR_CLOCK_REALTIME